### PR TITLE
Add HAPI Server with Synthetic data for testing

### LINKED ISSUE #1583: Set up HAPI server with docker compose and dummy data for testing
In order to prepare for writing integration and end to end tests that do not depend on external FHIR APIs let's set ourselves up with some standard test patients (maybe extracted from the existing sandboxes) and an easy way to load this data into a containerized instance the HAPI server.

Acceptance Criteria
GIVEN some synethic FHIR data
WHEN a docker compose file is used
THEN an instance of HAPI seeded with the synthetic data and the TEFCA Viewer are spun up

This should create a local development environment complement with access to synthetic data via a FHIR API for dev and testing.

### PR DESCRIPTION
**Pull request by m-goggins**
# PULL REQUEST

## Summary
This PR adds functionality to spin up a HAPI FHIR server and load it with synthetic data for testing. I add this functionality in `docker-compose-dev.yaml` because I think we only want to spin up this FHIR server for testing and not when we build the actual service for production. I'm happy to change this set up back to all in one docker compose file if folks feel differently because users of the service would connect to their own server/QHIN, not the HAPI server. 

To spin up the dev environment, run `docker compose -f .\docker-compose-dev.yaml up` from within the tefca container. It takes a second for the server to spin up and another 2-3 minutes for the data to populate in the server. You can check that everything works by going to `localhost:8080/fhir/Patient` in your browser; you should expect to see 3 Patient resources available if the server is spun up and contains the synthetic data. 

Shoutout to @alismx for the help debugging yesterday!  

## Related Issue
Fixes #1583 

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Review by robertandremitchell — COMMENTED** [review]
having some issues with the HAPI server initializing
```bash
tefca-fhir-server-1  |  ✓ Ready in 4.6s
tefca-data-loader-1  | Server is not healthy after 120 seconds.
tefca-data-loader-1 exited with code 1
```

I am exploring the hapi stuff but not having much luck. I'll review again tomorrow / can give more context if it would be helpful.

**Review by robertandremitchell — COMMENTED** [review]
I _finally_ got a version to work! However, I needed to increase the memory size of Java. 

```yaml
  tefca-fhir-server:
    build:
      context: .
      dockerfile: Dockerfile.dev
    image: "hapiproject/hapi:latest"
    ports:
      - "8080:8080"
    environment:
      - JAVA_OPTS=-Xmx2g -Xms512m
    command:
      [
        "sh",
        "-c",
        "ulimit -c unlimited && java $JAVA_OPTS -jar /app/main.war"
      ]
    depends_on:
      - tefca-viewer
```

however, even this is proving really finnicky to re-spin up, I really am at a loss for why it's proving so difficult to spin up. So I think you could not add this, especially if I'm the only one who's having this issue, but it's possible we need to add code to explicitly install and set a java version to the Dockerfile maybe? that was the only other recommendation I came across beyond just increasing the memory

**Comment by alismx** [comment]
> I _finally_ got a version to work! However, I needed to increase the memory size of Java.
> 
> ```yaml
>   tefca-fhir-server:
>     build:
>       context: .
>       dockerfile: Dockerfile.dev
>     image: "hapiproject/hapi:latest"
>     ports:
>       - "8080:8080"
>     environment:
>       - JAVA_OPTS=-Xmx2g -Xms512m
>     command:
>       [
>         "sh",
>         "-c",
>         "ulimit -c unlimited && java $JAVA_OPTS -jar /app/main.war"
>       ]
>     depends_on:
>       - tefca-viewer
> ```
> 
> however, even this is proving really finnicky to re-spin up, I really am at a loss for why it's proving so difficult to spin up. So I think you could not add this, especially if I'm the only one who's having this issue, but it's possible we need to add code to explicitly install and set a java version to the Dockerfile maybe? that was the only other recommendation I came across beyond just increasing the memory

To solve this, I'd try increasing the total amount of memory that docker can consume.

**Review by robertmitchellv — APPROVED** [review]
forgot to approve yesterday after testing locally that it works; great job!

**Review by robertandremitchell — APPROVED** [review]
thanks for the suggestion @alismx !

This seems to have done the trick, though, I still had to pull the image before getting it to work. Not sure if we should explicitly add a `FROM` to the dockerfile.dev to pull the image. If it already worked for everyone else, it's probably fine.

File: containers/tefca-viewer/docker-compose-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "3000:3000"
+    environment:
+      NODE_ENV: development
+  tefca-fhir-server:
+    build:
+      context: .
+      dockerfile: Dockerfile.dev
```
Comment (alismx) [inline containers/tefca-viewer/docker-compose-dev.yaml:13]: I think this file is being used for the `tefca-data-loader` service, not for this one, I think this build context should be removed. I think that'll solve @robertandremitchell issue.

**Review by alismx — CHANGES_REQUESTED** [review]

File: containers/tefca-viewer/docker-compose-dev.yaml (hunk 1 above)
Comment (m-goggins) [reply containers/tefca-viewer/docker-compose-dev.yaml]: Good catch, thanks! I just fixed. @robertandremitchell will you let me know if you're still having the same issue?

Comment (robertandremitchell) [reply containers/tefca-viewer/docker-compose-dev.yaml]: thanks! That did the trick for me 😁 
Sorry for the journey on this one 😅

Comment (m-goggins) [reply containers/tefca-viewer/docker-compose-dev.yaml]: Yay! Thank you both @alismx @robertandremitchell!

**Review by alismx — APPROVED** [review]

**Pull request merged**